history-service/model/repository.js

```diff
@@ -0,0 +1,21 @@
+import HistoryModel from './history-model.js';
+import 'dotenv/config';
+
+//Set up mongoose connection
+import mongoose from 'mongoose';
+
+let mongoDB = process.env.ENV == "PROD" ? process.env.CLOUD_DB_URL : process.env.LOCAL_DB_URL;
+
+mongoose.connect(mongoDB, { useNewUrlParser: true, useUnifiedTopology: true });
+
+let db = mongoose.connection;
+db.on('error', console.error.bind(console, 'MongoDB connection error:'));
+
+
+export async function getQuestionHistory(userName) {
+    return HistoryModel.find({ userName: userName }).select('title topic difficulty -_id');
+}
+
+export async function addQuestionToHistory(data) {
```

Comment:
If a user attempts the same question multiple times, we will have multiple entries that will be returned to the frontend and appear like a log of the attempts. Is this intended?

Reply:
now each question for a user will only be recorded once(last attempted) even if he does the same question multiple times.